docs/services/invocation.md

```diff
@@ -141,38 +141,60 @@ The Invocation identifier is opaque and its current format should not be relied
 
 ## Cancel an invocation
 
-:::caution
+If an invocation takes too long to complete or is no longer of interest, you can cancel it.
+Canceling an invocation allows it to free any resources it is holding and roll back any changes it has made so far.
+In order to roll back changes correctly, the service handlers need to contain the necessary compensation logic.
+If the required compensation logic is implemented, then the service state stays consistent even in the presence of cancellations.
 
-At the moment, gracefully cancelling an invocation is not supported. It will be supported in future Restate releases.
+The cancellation process works recursively in the following way:
+First, Restate tries to cancel the leaves of the current invocation, i.e. interrupt ongoing sleeps and awakeables or try to cancel calls to other services.
+Once the leaves are canceled, a terminal error is thrown in the service handler at the point in the code that the invocation had reached.
+This allows the handler to run its specific compensation logic.
+The response of the handler will then be propagated back to its caller where it will continue with the cancellation process.
 
+:::note
+Canceling an invocation is a non-blocking operation. This means that the cancellation is not guaranteed to have completed when the API call returns. In some rare cases, cancellations will not have an effect. In these cases, users need to retry the operation.
 :::
 
+```shell
+$ curl -X DELETE <RESTATE_META_ENDPOINT>/invocations/<INVOCATION_IDENTIFIER>
```

Comment:
```suggestion
$ curl -X DELETE <RESTATE_META_URL>/invocations/<INVOCATION_ID>
```

There is a PR open to unify this across the docs

Reply:
Good catch will update it accordingly.